### PR TITLE
sys-apps/yarn: bump to version 1.15.2

### PR DESCRIPTION
**Pull request by saboya**

### CODE REVIEW

File: sys-apps/yarn/yarn-1.15.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline sys-apps/yarn/yarn-1.15.2.ebuild]: 7 should work here, please use it.

File: sys-apps/yarn/yarn-1.15.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+src_install() {
+	local install_dir="/usr/$(get_libdir)/node_modules/yarn" path shebang
+	insinto "${install_dir}"
+	doins -r .
+	dosym "../$(get_libdir)/node_modules/yarn/bin/yarn.js" "/usr/bin/yarn"
+
+	while read -r -d '' path; do
+		read -r shebang < ${path} || die
+		[[ "${shebang}" == \#\!* ]] || continue
+		chmod +x "${path}" || die #614094
```
Comment (juippis) [inline sys-apps/yarn/yarn-1.15.2.ebuild]: Could this be switched to fperms instead? How many files does this monster modify, can they be predicted and this cleaned?

**Comment by saboya** [comment]
I don't want to sound harsh or bratty, but I feel the need to point this out here:

The remarks you made on this PR were made the last time I tried to bump `sys-apps/yarn` to the latest version, here: https://github.com/gentoo/gentoo/pull/10364

I modified the ebuild to account for them, and I'm glad the Gentoo team actually reviews PRs and have high standards for contributions, to ensure a better experience for its users.

What doesn't help is that, 6 days after I opened the PR, and 5 days after I made modifications to conform to the standards asked, another dev bumped the package by simply copying and renaming the ebuild, basically what I attempted to do here. AS a Gentoo user, I'm happy I can use the version of the package that I wanted. As I contributor, this type of situation compels me not bother to attempt to contribute again and just keep my local overlay updated.

That was not the first time that happened, and it's probably known by now how hard it is to actually try to contribute to Gentoo. This is scaring away people that could be having a positive impact on the distro that we love so much.

I will make the modifications you asked. Thanks for using your time to make Gentoo a better distro.

File: sys-apps/yarn/yarn-1.15.2.ebuild (hunk 2 above)
Comment (saboya) [reply sys-apps/yarn/yarn-1.15.2.ebuild]: Nowadays, actually not too many:

```
$ equery files --tree yarn
 * Searching for yarn ...
 * Contents of sys-apps/yarn-1.15.2:
 /usr
   > /bin
      + yarn -> ../lib64/node_modules/yarn/bin/yarn.js
   > /lib64
      > /node_modules
         > /yarn
            + LICENSE
            + README.md
            > /bin
               + yarn
               + yarn.cmd
               + yarn.js
               + yarnpkg
               + yarnpkg.cmd
            > /lib
               + cli.js
               + v8-compile-cache.js
            + package.json
```

I modified it to use fperms anyway, keeping the loop. Could be done manually so it's more explicit in the ebuild.

**Comment by juippis** [comment]
Looks like zmedico and williamh need this package in their work environment. Maybe they could be added as co-maintainers, so they get notified when you try to maintain it? 

@zmedico could you look into this PR, and merge it if it looks good. Thanks.

**Comment by zmedico** [comment]
Merged in c6d20924bc15946b2840ad43cdc68fbc5aede91a, thanks!

**Comment by zmedico** [comment]
> I don't want to sound harsh or bratty, but I feel the need to point this out here:
> 
> The remarks you made on this PR were made the last time I tried to bump `sys-apps/yarn` to the latest version, here: #10364
> 
> I modified the ebuild to account for them, and I'm glad the Gentoo team actually reviews PRs and have high standards for contributions, to ensure a better experience for its users.
> 
> What doesn't help is that, 6 days after I opened the PR, and 5 days after I made modifications to conform to the standards asked, another dev bumped the package by simply copying and renaming the ebuild, basically what I attempted to do here. AS a Gentoo user, I'm happy I can use the version of the package that I wanted. As I contributor, this type of situation compels me not bother to attempt to contribute again and just keep my local overlay updated.
> 
> That was not the first time that happened, and it's probably known by now how hard it is to actually try to contribute to Gentoo. This is scaring away people that could be having a positive impact on the distro that we love so much.
> 
> I will make the modifications you asked. Thanks for using your time to make Gentoo a better distro.

Your contributions are appreciated. In 9e39abab7efd38ae878211261529a47209286bbf I've added myself as a maintainer so that I'll be notified of future PRs that you send for sys-apps/yarn.

> Looks like zmedico and williamh need this package in their work environment. Maybe they could be added as co-maintainers, so they get notified when you try to maintain it?

I've added myself to maintainers, thanks!

> @zmedico could you look into this PR, and merge it if it looks good. Thanks.

Yep, merged in c6d20924bc15946b2840ad43cdc68fbc5aede91a.

**Pull request closed without merging**